### DISCUSSION TITLE
Center 301 connection to Artisan

### DISCUSSION THREAD

**Discussion by Goss111** [artisan-roaster-scope/artisan#770] [General]
Team,
I am trying to connect my Huky to Artisan. Through a Center 301 thermometer. 
I am using the “Connect Device” menu and selecting the Center 301.
I am then presented with a number of options to select. Including Port selection and a number of different options. 
Cannot get anything to work.
Searched everything that I can find and stuck.
Thanks in advance.
Hope I can figure this out.
Martin

**Comment by poundy** [comment]
Simplest answer is you've selected the wrong or an inoperable serial port on your PC.
According to my previous research, the Center 301 is a dual port thermocouple device, and will require the use of a USB-to-RS232 adapter.  I've also seen many people report that generic USB-RS232 adapters are not overly successful and suggest using one from Mr Li (the Huky guy) as he has a source of reliable ones.    
Walk us thru more of the detail on what you have done. Specifically, how you decided what COM port you were going to set up in Artisan, and how you knew it was the correct one?  
Are you on PC or MAC?

**Reply by Goss111** [reply]
Thanks for the assistance.  
I am a Mac user running the Artisan Software on a Dell 582O Windows Laptop.
Your research is correct.
The Center 301 requires the adaptor. I procured it directly from Kuanho Li when I bought the Huky. 
I have installed the Artisan Software.
I believe I got the right Driver. But that was not the easiest process as I know very little about the concept of Win Drivers.
I installed the Drivers.
Now I will attach the Huky to the 301. Readout on the 301 shows correct temp for both wires.
I then plug the 301 into the laptop.
Laptop recognizes the connection.
Then I launch the Artisan software.
Click "Config Device". Click "Center 301". Menu comes up and I click through it.
Then I get the Port menu. Default value is Com 4. This must be where it goes poorly. Default value if Com 4. I click through that. Artisan shows nothing connected. Nothing more than the green "reset" button in the upper right.
I am unable to figure out what to do next. I have no ability yet to determine com ports or assign them.
On a positive note I have already produced better roasts with the Center 301 readout then I have produced in the 15 years that I roasted with the Behmor.
I hope I can figure it out with my limited capability.
Thanks again for the help.
Very much appreciated.
Cheers,

**Reply by poundy** [reply]
all good - thanks for filling in the extra details, it's very helpful.

The only gap we have is making sure that the serial device is seen at a Windows layer, and making sure that it is known as COM4.  Once we can confirm that, we can configure in Artisan and it should all work.

First step here is opening Device Manager. Hit Windows key, type "device manager" with no quotes, and open the app. Make sure your USB-RS232 adapter is connected to the laptop and the meter and the meter is on.  

In DeviceManager, if you have set up everything for your variant of connector, you should see a section with COM PORTS that is visible, but is in collapsed mode.  If that's the case, expand that node, and then you'll see the actual COM ports that you machine has visible - if it's only one, then good chance it's the one for the USB-RS adapter, but to check you should remove it from the laptop and you should see it disappear.  Plug back in and make sure it comes back with the same COM number (often if you select a different USB port to connect it into, you'll see a different COM number, so I'd always suggest picking the same port every time)

 Here's a snip of DevMgr on my PC with no serial port attached: 
![image](https://user-images.githubusercontent.com/279966/146866814-7559ee52-9609-4b46-b5ed-a04d7e2893f7.png)

Here it is with the serial port in and the correct driver loaded - in collapsed view then expanded.
![image](https://user-images.githubusercontent.com/279966/146866910-7042572f-031d-41e9-ae9b-11edd8dd8c6c.png) 
![image](https://user-images.githubusercontent.com/279966/146866983-33da46bc-31ac-4152-9848-e76f36793483.png)

So from there, you can see in my case the driver is loaded correctly (no exclaimation mark against it) and it's a Silicon Labs CP210x driver registered on COM3.  

The aim here is to make sure you get a new device appear when you connect the USB-RS cable, that it has a valid driver loaded, and that you know what COM port it shows.... then, try that in your Artisan setup and see how we go.

**Comment by Goss111** [comment]
You Sir?
You Sir are the MAN!
The Flux Capacitor is now getting the full 1.21 Jigawatts.
Reinstalled the driver and designated on Artisan that the device is on COM3.
I very much appreciate you assistance.
Thank you.
Merry!

**Reply by poundy** [reply]
great to hear - now go make some even more awesome coffee !
